extension/sidebar-pane.js

```diff
@@ -0,0 +1,206 @@
+  const predicateEl = document.createElement("span");
+
+  if (invalid) {
+    predicateEl.appendChild(_renderLabel(" is invalid."));
+    return predicateEl;
+  }
+
+  const warningEl = _renderWarning(issue);
+  if (warningEl) {
+    predicateEl.appendChild(warningEl);
+  }
+
+  if (unsupportedBrowsers) {
+    if (warningEl) {
+      predicateEl.appendChild(_renderLabel(" and "));
+    }
+
+    const auxiliaryVerb =
+      type === WebCompat.ISSUE_TYPE.CSS_PROPERTY || aliases.length === 1 ? "does" : "do";
+    predicateEl.appendChild(_renderLabel(` ${ auxiliaryVerb } not support`));
+    predicateEl.appendChild(_renderBrowsersElement(unsupportedBrowsers));
+  }
+
+  predicateEl.appendChild(_renderLabel("."));
+  return predicateEl;
+}
+
+function _renderWarning(issue) {
+  const { type, aliases, deprecated, experimental } = issue;
+
+  if (!deprecated && !experimental) {
+    return null;
+  }
+
+  const wanringEl = document.createElement("span");
+
+  const connection =
+    type === WebCompat.ISSUE_TYPE.CSS_PROPERTY || aliases.length === 1 ? " is " : " are ";
+  wanringEl.appendChild(_renderLabel(connection));
+
+  if (deprecated) {
+    const deprecatedEl = _renderLabel("deprecated");
+    deprecatedEl.classList.add("deprecated");
+    wanringEl.appendChild(deprecatedEl);
+  }
+
+  if (experimental) {
+    const experimentalEl = _renderLabel("experimental");
+    experimentalEl.classList.add("experimental");
+
+    if (deprecated) {
+      wanringEl.appendChild(_renderLabel(" and "));
+    }
+
+    wanringEl.appendChild(experimentalEl);
+  }
+
+  return wanringEl;
+}
+
+function _renderBrowsersElement(browsers) {
+  const browsersEl = document.createElement("span");
+
+  const map = {};
+  for (const { brandName, name, version } of browsers) {
+    if (!map[name]) {
+      map[name] = { brandName, versions: [] };
+    }
+    map[name].versions.push(version);
+  }
+
+  for (let name in map) {
+    const { brandName, versions } = map[name];
+    const browserEl = _renderLabel(brandName);
+    browserEl.classList.add("browser");
+    browserEl.classList.add(name);
+
+    const versionsEl = document.createElement("span");
+    versionsEl.classList.add("versions");
+
+    for (const version of versions) {
+      const versionEl = _renderLabel(version);
+      versionEl.classList.add("version");
+      versionsEl.appendChild(versionEl);
+    }
+
+    browserEl.appendChild(versionsEl);
+    browsersEl.appendChild(browserEl);
+  }
+
+  return browsersEl;
+}
+
+function _renderTerms(terms, classes) {
+  const containerEl = document.createElement("span");
+
+  for (const term of terms) {
+    const termEl = _renderLabel(term);
+    termEl.classList.add(...classes);
+    containerEl.appendChild(termEl);
+  }
+
+  return containerEl;
+}
+
+function _renderLabel(text) {
+  const labelEl = document.createElement("label");
+  labelEl.textContent = text;
+  return labelEl;
+}
+
+function _getTargetBrowsers() {
```

Comment:
Yeah, indeed, moving the method into `WebCompat` class is making sense for now.
However, as you said, since I want to make the target browser configurable in near future, please let me locate in sidebar-pane.js as it is.  But will get them only once in loading this js.

Reply:
Sure thing! I initially thought this could live in `WebCompat`, but then I remembered seeing the options page of the full extension where users can configure the list of browsers they're interested in. So it certainly makes sense to keep this in `siderbar-pane.js` and outside the `WebCompat` library. 

Thanks for extracting its result to a constant computed only once. This is helpful!